matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/user/DefaultUserService.kt

```diff
@@ -36,18 +35,10 @@ internal class DefaultUserService @Inject constructor(private val userDataSource
         return userDataSource.getUser(userId)
     }
 
-    override suspend fun resolveUser(userId: String): User {
-        val known = getUser(userId)
-        if (known != null) {
-            return known
-        } else {
-            val params = GetProfileInfoTask.Params(userId)
-            val data = getProfileInfoTask.execute(params)
-            return User(
-                    userId,
-                    data[ProfileService.DISPLAY_NAME_KEY] as? String,
-                    data[ProfileService.AVATAR_URL_KEY] as? String)
-        }
+    override suspend fun resolveUser(userId: String) = getUser(userId) ?: run {
```

Comment:
When reading the code in GitHub, it's far better to keep explicit type `: User {`
Short `fun` form with `=` should be reserved for very small functions.